### PR TITLE
[mlir][LLVM] Use int32_t to indirectly construct GEPArg

### PR DESCRIPTION
**Pull request by andrey-golubev**
GEPArg can only be constructed from int32_t and mlir::Value. Explicitly cast other types (e.g. unsigned, size_t) to int32_t to avoid narrowing conversion warnings on MSVC. Some recent examples of such are:

mlir\lib\Dialect\LLVMIR\Transforms\TypeConsistency.cpp: error C2398: Element '1': conversion from 'size_t' to 'T' requires a narrowing conversion
    with
    [
        T=mlir::LLVM::GEPArg
    ]

mlir\lib\Dialect\LLVMIR\Transforms\TypeConsistency.cpp: error C2398: Element '1': conversion from 'unsigned int' to 'T' requires a narrowing conversion
    with
    [
        T=mlir::LLVM::GEPArg
    ]

### CODE REVIEW

**Comment by andrey-golubev** [comment]
@Dinistro @zero9178 @ThomasRaoux please take a look!

CC @joker-eph

**Comment by andrey-golubev** [comment]
Also, how do I backport this to 18.x properly? Should I retarget the PR to 18.x?

**Comment by Dinistro** [comment]
> Also, how do I backport this to 18.x properly? Should I retarget the PR to 18.x?

MLIR is not part of the release cycle, as far as I know. So no need to do that. Correct me if I'm wrong, @joker-eph.

File: mlir/lib/Dialect/LLVMIR/Transforms/TypeConsistency.cpp
Comment (Dinistro) [inline mlir/lib/Dialect/LLVMIR/Transforms/TypeConsistency.cpp:530]: Nit: It might be easier to just change the type of `currentOffset`, or even the `storeOffset` parameter.

**Review by Dinistro — APPROVED** [review]
Thanks for the change. Explicit casting does indeed make sense for these cases. LGTM % one nit.

For future PRs, just request review when you want a set of people to take a look.

**Comment by zero9178** [comment]
> Also, how do I backport this to 18.x properly? Should I retarget the PR to 18.x?

See https://llvm.org/docs/GitHub.html#backporting-fixes-to-the-release-branches and https://github.com/llvm/llvm-project/issues/79511 as an example. In short the process is to just land the change in `main`, open an issue, add it to the release milestone and use the cherry pick command. This will automatically create a PR to separately consider whether to backport the PR rather than just landing it in main.

**Review by zero9178 — APPROVED** [review]

File: mlir/lib/Dialect/LLVMIR/Transforms/TypeConsistency.cpp
Comment (andrey-golubev) [reply mlir/lib/Dialect/LLVMIR/Transforms/TypeConsistency.cpp:530]: it participates in other expressions involving `unsigned` values (e.g. as "accumulator" below) so figured the cast here is cleaner.

**Comment by andrey-golubev** [comment]
> For future PRs, just request review when you want a set of people to take a look.

Do you mean via GitHub "reviewers"? I think this is only possible for people who have commit access and I don't (yet) :| so have to ping you explicitly :p

**Comment by Dinistro** [comment]
> Do you mean via GitHub "reviewers"? I think this is only possible for people who have commit access and I don't (yet) :| so have to ping you explicitly :p

That's a bit odd, but what ever. Should we merge this for you then once the Windows build bot is done?

**Comment by Dinistro** [comment]
I'll merge this now, as the Windows build bot will probably never get a runner...

**Comment by andrey-golubev** [comment]
> I'll merge this now, as the Windows build bot will probably never get a runner...

Thank you! In fact, i think it will but after a while (iirc i had it finished after a day or so once).

**Pull request merged**
